core/src/main/java/kafka/server/TierStateMachine.java

```diff
@@ -230,6 +230,10 @@ private Long buildRemoteLogAuxState(TopicPartition topicPartition,
             }
         }
 
+        if (!rlm.isPartitionReady(topicPartition)) {
+            throw new RemoteStorageException(new IllegalStateException("RemoteLogManager is not ready for partition: " + topicPartition));
```

Comment:
Better to declare a defined `RuntimeException` for a partition not being ready instead of `IllegalStateException`.

Reply:
Introduced RetriableRemoteStorageException to gracefully handle the error. Thanks for the review!